update grpc endpoint and create namespace docs to include api keys (#3390)
temporalio/documentation: docs/production-deployment/cloud/account-setup/namespaces.mdx

Every Temporal Namespace uses a gRPC endpoint for communication.
When migrating to Temporal Cloud, you'll need to switch the gRPC endpoint in your code from your current hosting, whether self-hosted or locally-hosted, to Temporal Cloud.

> A gRPC endpoint is found on the detail page for each Cloud Namespace.

Reviewer: ```suggestion
A gRPC endpoint appears on the detail page for each Cloud Namespace.
```

:::tip

> By default, each account is allocated a limit of with ten Namespaces.

Reviewer: ```suggestion
By default, each account is allocated with a limit of ten Namespaces.
```

   When choosing this value, consider your needs for Event History versus the cost of maintaining that Event History.
   Typically, a development Namespace has a short retention period and a production Namespace has a longer retention period.
   (If you need to change this value later, contact [Temporal Support](/cloud/support#support-ticket).)

> 1. Select your authentication method between [API keys](/cloud/api-keys) or [mTLS](/cloud/certificates)

Reviewer: ```suggestion
1. Select your authentication method : [API keys](/cloud/api-keys) or [mTLS](/cloud/certificates).
```

> 1. If using the mTLS authentication method, paste the CA certificate for this Namespace.

Reviewer: ```suggestion
1. If using mTLS authentication, paste the CA certificate for this Namespace.
```

Before considering an appropriate Namespace configuration, you should be aware of the following constraints:

> - By default, each account is allocated a limit of ten Namespaces.

Reviewer: ```suggestion
- By default, each account is allocated with a limit of ten Namespaces.
```

- A Namespace is a security isolation boundary.
  Access to Temporal by [Worker Processes](/workers#worker-process) is permitted at the Namespace level.
  Isolating applications or environments (development, test, staging, production) should take this into consideration.

> - A Namespace is provisioned with an endpoint for executing your workflows.

Reviewer: ```suggestion
- A Namespace is provisioned with an endpoint for executing your Workflows.
```
